### PR TITLE
hdinsight custom ambari db template

### PR DESCRIPTION
**Pull request by tylerfox**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-hdinsight-custom-ambari-db/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    }
+  },
+  "variables": { 
+  	"sqldbname": "[concat(parameters('existingSQLServerName'), '/', parameters('newMetastoreDBName'))]",
```
Comment (bmoore-msft) [inline 101-hdinsight-custom-ambari-db/azuredeploy.json]: format the JSON (e.g. SHIFT+ALT+F in VS Code)

File: 101-hdinsight-custom-ambari-db/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+    "nestedtemplate_name": "[concat('nestedtemplate',uniqueString(resourceGroup().id))]"
```
Comment (bmoore-msft) [inline 101-hdinsight-custom-ambari-db/azuredeploy.json]: simplify this and just use a name that makes sense for the workload/deployment, eg. sqlDbDeployment

File: 101-hdinsight-custom-ambari-db/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+              },
+              "osProfile": {
+                "linuxOperatingSystemProfile": {
+                  "username": "[parameters('sshUserName')]",
+                  "password": "[parameters('sshPassword')]"
+                }
+              },
+              "virtualNetworkProfile": {
+                "id": "[resourceId(parameters('existingVirtualNetworkResourceGroup'), 'Microsoft.Network/virtualNetworks', parameters('existingVirtualNetworkName'))]",
+                "subnet": "[concat(resourceId(parameters('existingVirtualNetworkResourceGroup'), 'Microsoft.Network/virtualNetworks', parameters('existingVirtualNetworkName')), '/subnets/', parameters('existingVirtualNetworkSubnetName'))]"
```
Comment (bmoore-msft) [inline 101-hdinsight-custom-ambari-db/azuredeploy.json]: remove concat and just use resourceID

resourceId(parameters('existingVirtualNetworkResourceGroup'), 'Microsoft.Network/virtualNetworks/subnets', parameters('existingVirtualNetworkName'), parameters('existingVirtualNetworkSubnetName'))

File: 101-hdinsight-custom-ambari-db/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
```
Comment (bmoore-msft) [inline 101-hdinsight-custom-ambari-db/azuredeploy.json]: remove concat and just use resourceId

File: 101-hdinsight-custom-ambari-db/prereqs/prereq.azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+          "apiVersion": "2018-06-01-preview",
+          "dependsOn": [
+            "[resourceId('Microsoft.Sql/servers', variables('serverName'))]"
+          ],
+          "properties": {
+            "startIpAddress": "0.0.0.0",
+            "endIpAddress": "0.0.0.0"
+          }
+        },
+        {
```
Comment (bmoore-msft) [inline 101-hdinsight-custom-ambari-db/prereqs/prereq.azuredeploy.json]: how is this used?

File: 101-hdinsight-custom-ambari-db/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,276 @@
...
+            	"database-name": "[parameters('newMetastoreDBName')]",
+            	"database-user-name": "[parameters('existingSQLServerUsername')]",
+            	"database-user-password": "[parameters('existingSQLServerPassword')]"
+            }
+          }
+        },
+        "storageProfile": {
+          "storageaccounts": [
+            {
+              "name": "[replace(replace(concat(reference(resourceId(parameters('existingClusterStorageResourceGroup'), 'Microsoft.Storage/storageAccounts/', parameters('existingClusterStorageAccountName')), '2018-02-01').primaryEndpoints.blob),'https:',''),'/','')]",
```
Comment (bmoore-msft) [inline 101-hdinsight-custom-ambari-db/azuredeploy.json]: remove extraneous concat()

File: 101-hdinsight-custom-ambari-db/azuredeploy.json (hunk 1 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-custom-ambari-db/azuredeploy.json]: Thanks. Addressed.

File: 101-hdinsight-custom-ambari-db/azuredeploy.json (hunk 2 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-custom-ambari-db/azuredeploy.json]: Thanks. Fixed in line 144.

File: 101-hdinsight-custom-ambari-db/azuredeploy.json (hunk 3 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-custom-ambari-db/azuredeploy.json]: fixed. Thanks

File: 101-hdinsight-custom-ambari-db/azuredeploy.json (hunk 4 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-custom-ambari-db/azuredeploy.json]: fixed in line 244. Thanks

File: 101-hdinsight-custom-ambari-db/azuredeploy.json (hunk 6 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-custom-ambari-db/azuredeploy.json]: fixed. Thanks

File: 101-hdinsight-custom-ambari-db/prereqs/prereq.azuredeploy.json (hunk 5 above)
Comment (AliciaLiMicrosoft) [reply 101-hdinsight-custom-ambari-db/prereqs/prereq.azuredeploy.json]: Fixed. Thanks

**Comment by hrasheed-msft** [comment]
@bmoore-msft Could you review the updates and see if they resolve all of the issues with the template?

**Comment by hrasheed-msft** [comment]
#sign-off

**Comment by hrasheed-msft** [comment]
@bmoore-msft Tyler has moved to a different role, but we are trying to complete the template that he initially created. Please advise how to proceed.

**Pull request merged**